calico/felix/fetcd.py

```diff
@@ -111,10 +118,20 @@ def __init__(self, config, hosts_ipset):
         self._resync_greenlet = gevent.spawn(self._periodically_resync)
         self._resync_greenlet.link_exception(self._on_worker_died)
 
+        # Create a client to write to etcd.
+        self.client = None
+        _reconnect(self)
```

Comment:
Good thinking in trying to keep the code DRY but passing self to a function is a big "code smell", especially if the function is accessing `self._xxx` (private) fields.

In this case, you could probably use the mix-in pattern or a wrapper/proxy object that delegates to the etcd client and holds the config.